generic/piranha-tree-sitter/src/test-resources/java/configurations/rules.toml

```diff
@@ -0,0 +1,453 @@
+#   
+#
+[[rules]]
+name = "delete_putToggleEnabled"
+query = """
+(
+(expression_statement (method_invocation name : (_) @name
+					arguments: (argument_list
+                    			([
+		                          (field_access field: (_)@argument)
+        		                  (_) @argument
+                		         ])))) @expression_statement
+ (#eq? @name "putToggleEnabled")
+ (#eq? @argument "@stale_flag_name")
+)
+"""
+replace_node = "expression_statement"
+replace = ""
+groups = []
+holes = ["stale_flag_name"]
+
+#
+# For @stale_flag_name = STALE_FLAG and @treated = true
+# Before :
+#   exp.putToggleDisabled(STALE_FLAG);
+# After :
+#   
+#
+[[rules]]
+name = "delete_putToggleDisabled"
+query = """
+(
+(expression_statement (method_invocation name : (_) @name
+					arguments: (argument_list
+                    			([
+		                          (field_access field: (_)@argument)
+        		                  (_) @argument
+                		         ])))) @expression_statement
+ (#eq? @name "putToggleDisabled")
+ (#eq? @argument "@stale_flag_name")
+)
+"""
+replace_node = "expression_statement"
+replace = ""
+groups = []
+holes = ["stale_flag_name"]
+
+#
+# For @stale_flag_name = STALE_FLAG and @treated = true
+# Before :
+#   exp.includeEvent(STALE_FLAG);
+# After :
+#   
+#
+[[rules]]
+name = "delete_includeEvent"
+query = """
+(
+(expression_statement (method_invocation name : (_) @name
+					arguments: (argument_list
+                    			([
+		                          (field_access field: (_)@argument)
+        		                  (_) @argument
+                		         ])))) @expression_statement
+ (#eq? @name "includeEvent")
+ (#eq? @argument "@stale_flag_name")
+)
+"""
+replace_node = "expression_statement"
+replace = ""
+groups = []
+holes = ["stale_flag_name"]
+
+# Before :
+# interface IExp {
+#  @BoolParam(key = "STALE_FLAG", namespace="NS")	
+#  public void isStaleFeature();
+#  @BoolParam(key = "OTHER_FLAG", namespace="NS")
+#  public void isOtherFeature();
+# }
+# After :
+# interface IExp {
+#  @BoolParam(key = "OTHER_FLAG", namespace="NS")
+#  public void isOtherFeature();
+# }
+#
+[[rules]]
+name = "delete_flag_method_declaration_fully_annotated"
+query = """
+(
+(method_declaration
+	 (modifiers (annotation
+     				name: (_) @annotation_name
+					arguments: (annotation_argument_list
+                    	(element_value_pair
+							 key: (_) @k1
+							 value: (_) @v1))))
+      name: (_) @m_name
+ )@method_declaration
+ (#eq? @annotation_name "BoolParam")
+ (#eq? @k1 "key")
+ (#eq? @v1 "\\\"@stale_flag_name\\\"")
+ )
+ (
+(method_declaration
+	 (modifiers (annotation
+					arguments: (annotation_argument_list
+                    	(element_value_pair
+							 key: (_) @k2
+							 value: (_) @v2))))
+ )@m2
+ (#eq? @k2 "namespace")
+ (#eq? @v2 "\\\"@namespace\\\"")
+ )
+"""
+replace_node = "method_declaration"
+replace = ""
+groups = ["Delete Method"]
+holes = ["stale_flag_name", "namespace"]
+
+# Before :
```

Comment:
Given this example, can we generate the tree sitter query given below automatically? Let us work on this as a follow up.